transmogrifier/sources/transformer.py

```diff
@@ -243,26 +242,26 @@ def _transform(
             timdex_record_id = self.get_timdex_record_id(
                 self.source, source_record_id, source_record
             )
-            raise DeletedRecord(timdex_record_id)
+            raise DeletedRecordError(timdex_record_id)
         optional_fields = self.get_optional_fields(source_record)
         if optional_fields is None:
             return None
-        else:
-            fields = {
-                **self.get_required_fields(source_record),
-                **optional_fields,
-            }
 
-            # If citation field was not present, generate citation from other fields
-            if fields.get("citation") is None:
-                fields["citation"] = generate_citation(fields)
-            if fields.get("content_type") is None:
-                fields["content_type"] = ["Not specified"]
+        fields = {
+            **self.get_required_fields(source_record),
+            **optional_fields,
+        }
+
+        # If citation field was not present, generate citation from other fields
+        if fields.get("citation") is None:
+            fields["citation"] = generate_citation(fields)
+        if fields.get("content_type") is None:
+            fields["content_type"] = ["Not specified"]
```

Comment:
This is a case where I think the removed `else` improves readability; kind of sets the default route as the happy path.